python/lsst/rubintv/production/rubinTv.py

```diff
@@ -1277,85 +1420,28 @@ def processDay(self, dayObs, skipEventNumbers={}):
-                clippedEl = clipDataToEvent(elevationData, event)
-
-                md = {}
-                azStart = None
-                elStart = None
-                azMove = None
-                elMove = None
-                maxElTorque = None
-                maxAzTorque = None
-
-                if len(clippedAz) > 0:
-                    azStart = clippedAz.iloc[0]['actualPosition']
-                    azMove = clippedAz.iloc[-1]['actualPosition'] - azStart
-                    # key=abs gets the item with the largest absolute value but
-                    # keeps the sign so we don't deal with min/max depending on
-                    # the direction of the move etc
-                    maxAzTorque = max(clippedAz['actualTorque'], key=abs)
-
-                if len(clippedEl) > 0:
-                    elStart = clippedEl.iloc[0]['actualPosition']
-                    elMove = clippedEl.iloc[-1]['actualPosition'] - elStart
-                    maxElTorque = max(clippedEl['actualTorque'], key=abs)
-
-                # values could be None by design, for when there is no data
-                # in the clipped dataframes, i.e. from the event window exactly
-                md['Azimuth start'] = azStart
-                md['Elevation start'] = elStart
-                md['Azimuth move'] = azMove
-                md['Elevation move'] = elMove
-                md['Largest azimuth torque'] = maxAzTorque
-                md['Largest elevation torque'] = maxElTorque
-
-                rowData = {event.seqNum: md}
-                writeMetadataShard(self.shardsDirectory, event.dayObs, rowData)
-
-                commands = getCommandsDuringEvent(self.client, event, self.commandsToPlot, doLog=False)
-                if not all([time is None for time in commands.values()]):
-                    rowData = {event.seqNum: {'Has commands?': '✅'}}
-                    writeMetadataShard(self.shardsDirectory, event.dayObs, rowData)
-
-                plotEvent(self.client,
-                          event,
-                          fig=self.figure,
-                          prePadding=self.prePadding,
-                          postPadding=self.postPadding,
-                          commands=commands,
-                          azimuthData=azimuthData,
-                          elevationData=elevationData,
-                          )
-
-                filename = self._getSaveFilename(dayObs, event)
-                self.figure.savefig(filename)
-                self.uploader.uploadPerSeqNumPlot(self.plotChannelName,
-                                                  dayObsInt=dayObs,
-                                                  seqNumInt=event.seqNum,
-                                                  filename=filename,
-                                                  isLiveFile=False
-                                                  )
+                rowData = self.runMountMotionAnalysis(event)
```

Comment:
I suspect this is the "unscalable plots" you were talking about.  These could be merged into a helper function, and probably should be if the number of plots expands much beyond 2.

Reply:
Yeah, exactly. The whole thing needs a refactor: DM-42911